ietf/utils/xmldraft.py

```diff
@@ -60,17 +60,43 @@ def cleanup():  # called when context exited, even if there's an exception
                 tree.tree = v2v3.convert2to3()
         return tree, xml_version
 
-    def _document_name(self, anchor):
-        """Guess document name from reference anchor
-
-        Looks for series numbers and removes leading 0s from the number.
-        """
-        anchor = anchor.lower()  # always give back lowercase
-        label = anchor.rstrip('0123456789')  # remove trailing digits
-        if label in ['rfc', 'bcp', 'fyi', 'std']:
-            number = int(anchor[len(label):])
-            return f'{label}{number}'
-        return anchor
+    def _document_name(self, ref):
+        """Get document name from reference."""
+        series = ["rfc", "bcp", "fyi", "std"]
+        # handle xinclude first
+        # FIXME: this assumes the xinclude is a bibxml href
```

Comment:
What's to fix here? If it's not a bibxml href, the code won't crash. Worst case it fails and reports a reference that's not really a reference, yes? Given that this is only called with blobs taken out of the references section, I'm having a hard time constructing a likely scenario that this could happen.

Reply:
(I am still a little itchy about looking at the href url form rather than expanding the xinclude and parsing the resulting xml instead, but also dont really want to add fetching external things at this point in the code.)

Reply:
The FIXME was exactly on that, i.e., that the proper wat of doing it would be to expamnd the xinclude rather than relying on a certain format in the URL. So there could still be false negatives.